teps/0145-cel-in-whenexpression.md

```diff
@@ -0,0 +1,372 @@
+              image: ubuntu
+              script: 'echo finally done'
+  params:
+    - name: path
+      value: README.md
+    - name: branches
+      value:
+        - main
+        - hotfix
+  workspaces:
+    - name: source
+      volumeClaimTemplate:
+        spec:
+          accessModes:
+            - ReadWriteOnce
+          resources:
+            requests:
+              storage: 16Mi
+```
+
+## Design Evaluation
+
+### Simplicity
+
+The proposal helps to reduce the verbosity in Tekton by using 1 string of CEL expression instead of 3 (input+operator+values).
+
+### Flexibility
+
+CEL is a powerful expression language that can be used to perform a wide variety of tasks in CI/CD pipelines. This increases the flexibility of `WhenExpression` in Tekton to handle more complex conditions.
+
+### Conformance
+
+The proposal will introduce an additive change to the current API. It does not impact conformance in the inital release.
+
+### User Experience
+
+This proposal should help to improve the user experience by
+- simplifying the syntax for `WhenExpression`
+- support more complex conditional execution instead of hacky workaround
+
+### Drawbacks
+
+Users need to be familar with CEL expression syntax to use it properly, this may be a learning curve but they can choose to use current `WhenExpression` syntax for simple cases.
+
+
+## Alternatives
+
+### Support More Operators in current WhenExpression
+Tekton can support more operators from [k8s apimachinery][k8s apimachinery] in `WhenExpression`'s `Operator`.
+
+### Use languages other than CEL
+CEL has been adopted in Tekton Triggers and CustomTask. We could add it first and consider other languages support as future work.
+
+## Implementation Plan
+
+### Update existing fields in v1 API
+
+Add `CEL` to `WhenExpression`:
+```go
+type WhenExpression struct {
+	// Input is the string for guard checking which can be a static input or an output from a parent Task
+	Input string `json:"input"`
+
+	// Operator that represents an Input's relationship to the values
+	Operator selection.Operator `json:"operator"`
+
+	// Values is an array of strings, which is compared against the input, for guard checking
+	// It must be non-empty
+	// +listType=atomic
+	Values []string `json:"values"`
+
+    // CEL is a CEL expression to evaluate, if follows the cel-spec language definition
+	CEL string `json:"cel",omitempty`
+}
+```
+
+### Upgrade and Migration Strategy
+
+This feature will be introduced in alpha with a dedicated feature flag which is disabled by default.
+
+When the feature is promoted to beta, the dedicated feature flag will be enabled by default.
+
+When the feature is promoted to stable, the dedicated feature flag will be removed and the feature will be available by default.
+
+### Implementation Pull Requests
+
+TODO
+
+## References
+- [Tekon Trigger][Trigger]
+- [CELCustomRun][CELCustomRun]
+- [cel-go][cel-go]
+
+[Trigger]: https://tekton.dev/docs/triggers/cel_expressions/
```

Comment:
All these references are missing `-` and hence not showing up in https://github.com/tektoncd/community/blob/739fffd0e7303dbef25c861e32cd7b57eb0c0107/teps/0145-cel-in-whenexpression.md#references unless that's the intention.

Reply:
Yes this is intention. e.g. we don't need to add https://tekton.dev/docs/triggers/cel_expressions/ link to everywhere we use Trigger, we can just use [Trigger](Trigger) to link it